[CJerrong] iP

CJerrong opened the pull request:
## Haro

> “I love lists. Always have.” – Adam Savage ([source](https://medium.com/@oaklinejournal/25-quotes-from-successful-people-on-the-power-of-writing-to-do-lists-8a0b3729a34c))


Haro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use
All you need to do is,

1. download it from [here](https://github.com/CJerrong/ip/releases).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
---
If you are a Java programmer, you can use it to practice Java too. Here's the ``main`` method:
``` java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

CJerrong committed:
Level 0:Rename, Greet, Exit

CJerrong committed:
Level-1: Echo

CJerrong committed:
Level-2: Add, List

CJerrong committed:
Level-3: Mark as Done

CJerrong committed:
Level-4: ToDos, Events, Deadlines

CJerrong committed:
A-TextUiTesting: Automated Text UI Testing

CJerrong committed:
Level-5: Handle Errors

CJerrong committed:
Level-6: Delete

CJerrong committed:
Add Storage class

CJerrong committed:
Add saveToDisk() method in Storage class

CJerrong committed:
Add data to .gitignore

CJerrong committed:
Add savedList.txt to gitignore

CJerrong committed:
Remove saveList.txt from cache

CJerrong committed:
Merge branch 'branch-Level-7'

CJerrong committed:
Add functionality to store deadline, event start and end as LocalDate

CJerrong committed:
Merge branch 'branch-Level-8'

CJerrong committed:
Refactor classes to have better OOP

CJerrong committed:
Refactor Storage class to return ArrayList instead of TaskList

CJerrong committed:
Merge branch 'branch-A-MoreOOP'

CJerrong committed:
Refactor classes into respective packages

CJerrong committed:
Merge branch 'branch-A-Packages'

CJerrong committed:
Merge branch 'add-gradle-support'

CJerrong committed:
Add Gradle build functionality to project

CJerrong committed:
Merge branch 'branch-A-Gradle'

CJerrong committed:
Add JUnit tests for TaskList and Parser

CJerrong committed:
Merge branch 'branch-A-JUnit'

CJerrong committed:
Remove redundant code in haro and taskList

CJerrong committed:
Add JavaDocs for public classes and methods

CJerrong committed:
Refactor code to abide by coding standards

CJerrong committed:
Add feature for users to find tasks by searching for keywords

CJerrong committed:
Merge branch 'branch-A-JavaDoc'

CJerrong committed:
Merge branch 'branch-A-CodingStandard'

CJerrong committed:
Merge branch 'branch-Level-9'

jeremiahlzz commented on .gitignore:
```diff
@@ -15,3 +15,6 @@ bin/
 /text-ui-test/ACTUAL.TXT
 text-ui-test/EXPECTED-UNIX.TXT
+
+/data/
+data/saveList.txt
```
`/data/` includes all files and directories within the `data` file.
Consider removing the redundant `data/saveList.txt`, or removing `/data/` to only exclude `saveList.txt` from commits.

jeremiahlzz commented on src/main/java/haro/Parser.java:
```diff
@@ -0,0 +1,214 @@
+
+            FindCommand findCommand = new FindCommand(commandArg);
+            resultCommand = findCommand;
+        }
```
The `switch case` and the `if-else` statements both accomplish the same thing, differentiate the instruction to be executed.
Consider using `Instruction.valueOf(instructWord.toUpperCase())` to convert the `String` input into an `Enum` type directly.

Should each instruction code also be abstracted out (as methods) to improve readability?

jeremiahlzz commented on src/main/java/haro/Storage.java:
```diff
@@ -0,0 +1,167 @@
+                        }
+
+                        int markedInt = Integer.valueOf(storageArgs[1].trim());
+                        boolean marked = markedInt == 0 ? false : true;
```
According to the coding standards, ternary expressions require braces around the boolean expression. This improves readability. For example:
```java
alpha = (aLongBooleanExpression) ? beta : gamma;
alpha = (aLongBooleanExpression)
        ? beta
        : gamma;
```

jeremiahlzz commented on src/main/java/haro/Storage.java:
```diff
@@ -0,0 +1,167 @@
+                        }
+
+                        int markedInt = Integer.valueOf(storageArgs[1].trim());
+                        boolean marked = markedInt == 0 ? false : true;
```
Ternary expression needs braces for the boolean expression.

jeremiahlzz commented on src/main/java/haro/Storage.java:
```diff
@@ -0,0 +1,167 @@
+                }
+            }
+
+        } catch (NoSuchElementException e) {
+
+        } catch (Exception e) {
+
```
Empty catch blocks. Should a comment be included to declare that it was left empty on purpose? E.g. `// Do Nothing`

jeremiahlzz reviewed (commented):
Nice use of custom exceptions to handle erroneous user input!

CJerrong committed:
Add checkstyle file to detect coding violations

CJerrong committed:
Merge branch 'branch-A-Checkstyle'

CJerrong committed:
Add CheckStyle functionality

CJerrong committed:
Merge branch 'branch-A-CheckStyle'

CJerrong committed:
Add JavaFx Launcher

CJerrong committed:
Add a simple echo GUI

CJerrong committed:
Add controller for using FXML

CJerrong committed:
Refactor Parser, Ui, Haro classes to fit new UI

CJerrong committed:
Add new user and haro images

CJerrong committed:
Merge branch 'branch-Level-10'

CJerrong committed:
Edit build.gradle to generate new JAR file

CJerrong committed:
Add assertions internal object validity in Parser and Storage

CJerrong committed:
Reformat code to align with coding standards

Boolean names: make boolean variables more boolean sounding

Boolean names are inconsistent

Renamed boolean names to be more boolean sounding, ie marked -> isMarked

Task List: Make methods more consistent with SLAP principle

Task List has inconsistent levels of abstraction

Added a new method to format task strings, thus methods in task list can
now align with SLAP principles

Ternary expressions: Make ternary expressions follow format in coding
standards

Ternary expressions do not follow format speicified in coding standards

Added braces around boolean expressions of ternary expressions to be in
line with coding standards

CJerrong committed:
Merge pull request #2 from CJerrong/branch-A-Assertions

A-Assertions: Use assertions

CJerrong committed:
Merge pull request #3 from CJerrong/branch-A-CodeQuality

A-CodeQuality: Improve code quality

CJerrong committed:
Add edit command for user to edit previously added tasks

CJerrong committed:
Merge branch 'branch-BCD-Extension'

CJerrong committed:
Add background colors and image crop

CJerrong committed:
Fix dialogBox expanding issue, dialogBox no longer cuts off text

CJerrong committed:
Refine GUI visual aspects

CJerrong committed:
Refactor haro to save after each user input

CJerrong committed:
Refactor Parser to reduce long methods

CJerrong committed:
Add javadocs for new methods in Parser

CJerrong committed:
Refactor mainWindow styling to resize the scrollpane with the window

CJerrong committed:
Add red color to error messages

CJerrong committed:
Refactor Storage loadSave() method to reduce method length

CJerrong committed:
Add title and icon for GUI window

CJerrong committed:
Edit GUI panel colors and add ui demonstration picture

CJerrong committed:
Merge branch 'branch-A-BetterGui'

CJerrong committed:
Refine colors of GUI

CJerrong committed:
Update ReadMe to include basic features

CJerrong committed:
Update README with more commands and update Ui.png

CJerrong committed:
Update README

CJerrong committed:
Merge branch 'branch-A-UserGuide'

CJerrong committed:
Refactor code base for more organisation

CJerrong committed:
Refactor styling to adhere with coding standards

CJerrong committed:
Refactor parser to handle edge case exceptions

CJerrong committed:
Refactor code to eliminate bugs in edit command

CJerrong committed:
Refine README

CJerrong committed:
Update README

CJerrong committed:
Update README minor fixes

CJerrong committed:
Update README
